config/env.config.js

```diff
@@ -69,26 +71,38 @@ const RedirectPlugin = function(options) {
                             if (match && match.length > 1) {
                                 return match[1];
                             }
-                            return "";
+                            return null;
                         };
-                        var originUrl = decodeURIComponent(getParameterByName("originEnvironment"));
-                        var newUrl = window.localStorage.getItem("environmentConfig");
-                        if (newUrl && !window.location.href.match(newUrl) && originUrl !== newUrl) {
+                        var originUrl = getParameterByName("originEnvironment");
+                        var newUrl = getParameterByName("newEnvironment");
+                        if (newUrl) {
+                            newUrl = decodeURIComponent(newUrl);
+                            if (window.location.href.match(newUrl)) {
+                                window.localStorage.removeItem("environmentConfig");
```

Comment:
Added comments to clarify logic and  took your feedback into account from yesterday. Hopefully they make the general flow clear now.